### ISSUE TITLE
not working properly

### ISSUE THREAD

**Issue by shabi786** [AliasIO/wappalyzer#3202]
It worked for many websites but it is not working for small websites and newly created websites.

**Comment by AliasIO** [comment]
Can you please provide examples?

**Comment by davidlenz** [comment]
Dont know if this website is small or newly created, however it just won't work for me.

`wappalyzer http://hortec-berlin.de` will never finish for me (tested on different machines).

```
root@Wappalyzer:~# wappalyzer http://www.hortec-berlin.de --debug

wappalyzer | log | driver | Launching browser...
wappalyzer | log | page | Navigate to http://www.hortec-berlin.de/
 ```

and never does anything after.

**Comment by AliasIO** [comment]
Which version are you using? Your example works fine for me.

**Comment by davidlenz** [comment]
Huh, weird. I'm on version `6.0.12`. What's your version?

```
root@Wappalyzer:~# npm list -g
/usr/local/lib
└─┬ wappalyzer@6.0.12
  ├── languagedetect@2.0.0
  └─┬ puppeteer@2.1.1
    ├── @types/mime-types@2.1.0
    ├─┬ debug@4.1.1
    │ └── ms@2.1.2
    ├─┬ extract-zip@1.7.0
    │ ├─┬ concat-stream@1.6.2
    │ │ ├── buffer-from@1.1.1
    │ │ ├── inherits@2.0.4
    │ │ ├─┬ readable-stream@2.3.7
    │ │ │ ├── core-util-is@1.0.2
    │ │ │ ├── isarray@1.0.0
    │ │ │ ├── process-nextick-args@2.0.1
    │ │ │ ├── safe-buffer@5.1.2
    │ │ │ ├── string_decoder@1.1.1
    │ │ │ └── util-deprecate@1.0.2
    │ │ └── typedarray@0.0.6
    │ ├─┬ debug@2.6.9
    │ │ └── ms@2.0.0
    │ ├─┬ mkdirp@0.5.5
    │ │ └── minimist@1.2.5
    │ └─┬ yauzl@2.10.0
    │   ├── buffer-crc32@0.2.13
    │   └─┬ fd-slicer@1.1.0
    │     └── pend@1.2.0
    ├─┬ https-proxy-agent@4.0.0
    │ └── agent-base@5.1.1
    ├── mime@2.4.6
    ├─┬ mime-types@2.1.27
    │ └── mime-db@1.44.0
    ├── progress@2.0.3
    ├── proxy-from-env@1.1.0
    ├─┬ rimraf@2.7.1
    │ └─┬ glob@7.1.6
    │   ├── fs.realpath@1.0.0
    │   ├─┬ inflight@1.0.6
    │   │ └── wrappy@1.0.2
    │   ├─┬ minimatch@3.0.4
    │   │ └─┬ brace-expansion@1.1.11
    │   │   ├── balanced-match@1.0.0
    │   │   └── concat-map@0.0.1
    │   ├── once@1.4.0
    │   └── path-is-absolute@1.0.1
    └─┬ ws@6.2.1
      └── async-limiter@1.0.1
```

**Comment by picco** [comment]
I have a similar issue, http://hortec-berlin.de hangs as does https://bet365.ee for example.
Using version 6.1.1

**Comment by AliasIO** [comment]
This seems to be related to https://github.com/puppeteer/puppeteer/issues/4072. I've added a timeout to these calls so that it doesn't hang in 892fe1ac147421c50625c5ad081fb98670c49a67.

**Comment by picco** [comment]
Thank you for the quick response!
As the exception is thrown inside setTimeout() my calling code in a node script cannot catch it
Does it make sense to also call .error() prior to throwing?
Tested that this makes it possible for me to catch it by using on.error handler:
```
  timeout() {
    return new Promise(() =>
      setTimeout(() => {
        let error = new Error('The website took too long to respond')
        this.error(error)
        throw error
      }, this.options.maxWait)
    )
  }
```

**Comment by AliasIO** [comment]
Fixed in 4e8cf66677f361add360d2d7004c5532472e9763.
